cmake/SpectreSetupPython.cmake

```diff
@@ -11,6 +11,7 @@ set(SPECTRE_LINK_PYBINDINGS
   -Wl,--whole-archive
   PUBLIC PyBindings
   PUBLIC ${HDF5_LIBRARIES}
+  PUBLIC ${GSL_LIBRARIES}
```

Comment:
@nilsleiffischer says this doesn't need to be in this PR.

Reply:
Dropped the commit. I experimented with cubic spline interpolation which requires GSL, but that's unrelated to this PR.